### ISSUE TITLE
[REQUEST]: Document full list of auto-updated packages 

### ISSUE THREAD

**Issue by kpollich** [elastic/ingest-docs#1222]
### Description

Today, we include docs around Fleet's auto-update and auto-policy-upgrade logic here: https://www.elastic.co/guide/en/fleet/current/upgrade-integration.html#upgrade-integration-policies-automatically. However, we don't expose the full list of packages that are subject to this behavior by default in Fleet. We should include this list. 

### Resources

The list of packages is defined in https://github.com/elastic/kibana/blob/d4dc118ffac98ce132d9b4569fdf63ed0884383e/x-pack/plugins/fleet/common/constants/epm.ts#L70-L75 and includes

- Endpoint
- APM
- Synthetics
- Cloud Security Posture Management

I believe `system`, `elastic_agent`, and `fleet_server` are also subject to this behavior, but need to confirm by digging a bit deeper. 

### Collaboration

The documentation team will investigate the issue and create the initial content.

### Point of contact.

**Main contact:** @kpollich 

**Stakeholders:** @elastic/fleet

**Comment by kilfoyle** [comment]
Thanks for opening this @kpollich. I'll wait for your confirmation on the `system`, `elastic_agent`, and `fleet_server` packages, and then it should be a quick fix.

**Comment by kpollich** [comment]
Confirming the behavior above:

- Only `synthetics` is auto-installed on a fresh cluster
- `system` is installed when "Collect system logs and metrics" is enabled on an agent policy

![image](https://github.com/user-attachments/assets/d3c49814-e524-4843-9145-b75fbbd26c45)

- `elastic_agent` is installed when these agent monitoring options are enabled when creating an agent policy (they're under advanced options and enabled by default)

![image](https://github.com/user-attachments/assets/839f971d-a8a1-4d76-89fe-4fd02588a254)

- `fleet_server` is installed when setting up a Fleet Server via Fleet UI

All of these integrations can be opted into automatic policy upgrades, but it's disabled by default. So, we won't automatically upgrade your agents at the edge to the newest version of these integrations. 

![image](https://github.com/user-attachments/assets/d0ce700e-21ba-45f1-a109-07fdc12ba4d7)
![image](https://github.com/user-attachments/assets/4cac8e04-5210-48c1-abe3-cca0120eda88)
![image](https://github.com/user-attachments/assets/1797b9e5-7beb-4dff-9a6a-60d3e7352184)

`apm`, `synthetics`, and `cspm` on the other hand are opted into automatic policy upgrades by default, and this behavior cannot be disabled.

![image](https://github.com/user-attachments/assets/e9e88e5d-db81-489f-8568-0a3c7bcee502)
![image](https://github.com/user-attachments/assets/36a1a1b1-0ec2-4bc4-bb7f-c6bc11902914)
![image](https://github.com/user-attachments/assets/afd9cdb2-5135-4271-a817-a7f146a9a196)

tl;dr
- `system`, `elastic_agent`, `fleet_server`, and `endpoint` can _opt in_ to automatic policy upgrades
- `apm`, `synthetics`, and `cspm` cannot _opt out_ of automatic policy upgrades

**Comment by kilfoyle** [comment]
Closed by https://github.com/elastic/ingest-docs/pull/1260
